applications/integrationtests/mechanics/IntegrationTest.cpp

```diff
@@ -0,0 +1,219 @@
+    }
+}
+
+BOOST_AUTO_TEST_CASE(PolynomialIntegrationTet) {
+    // Tetrahedron
+    NuTo::IntegrationType3D4NGauss1Ip intGaussTet1;
+    NuTo::IntegrationType3D4NGauss4Ip intGaussTet4;
+
+    std::vector<NuTo::IntegrationTypeBase *> intTypesTet = {
+        &intGaussTet1, &intGaussTet4
+    };
+
+    // Analytical results (result[i] = integral of polynomial of order i with all
+    // coefficients set to 1)
+    std::vector<double> analyticalResultsTet = {
+            1./6.,
+            7./24.,
+            11./30.,
+            59./144.,
+            313./720.,
+            9067./20160.,
+            83317181440.
+    };
+
+    int polyOrder = 5;
+
+    std::cout << "----------------------------" << std::endl;
+    std::cout << "Expected Result Tet: " << analyticalResultsTet[polyOrder]
+                 << std::endl;
+
+    for (NuTo::IntegrationTypeBase *intType : intTypesTet) {
+        std::cout << "Int Points " << intType->GetNumIntegrationPoints()
+                  << "Result: " << integrate3D(polyOrder, *(intType)) << std::endl;
+    }
+}
+
+BOOST_AUTO_TEST_CASE(PolynomialIntegrationBrick) {
+    // 3D
+    NuTo::IntegrationTypeTensorProduct<3> intGaussBrick1(1,NuTo::eIntegrationMethod::GAUSS);
+    NuTo::IntegrationTypeTensorProduct<3> intGaussBrick2(2,NuTo::eIntegrationMethod::GAUSS);
+
+    NuTo::IntegrationTypeTensorProduct<3> intLobBrick3new(3,NuTo::eIntegrationMethod::LOBATTO);
+    NuTo::IntegrationTypeTensorProduct<3> intLobBrick4new(4,NuTo::eIntegrationMethod::LOBATTO);
+    NuTo::IntegrationTypeTensorProduct<3> intLobBrick5new(5,NuTo::eIntegrationMethod::LOBATTO);
+
+    std::vector<NuTo::IntegrationTypeBase *> intTypesBrick3D = {
+        &intGaussBrick1, &intGaussBrick2,
+        &intLobBrick3new, &intLobBrick4new, &intLobBrick5new
+    };
+
+    // Analytical results (result[i] = integral of polynomial of order i with all
+    // coefficients set to 1)
+    std::vector<double> analyticalResultsBrick = {8.,
+                                               8.,
+                                               16.,
+                                               16.,
+                                               352. / 15,
+                                               352. / 15,
+                                               5744. / 189,
+                                               5744. / 189,
+                                               174056. / 4725,
+                                               174056. / 4725};
+
+    int polyOrder = 5;
+
+    std::cout << "----------------------------" << std::endl;
+    std::cout << "Expected Result Brick: " << analyticalResultsBrick[polyOrder]
+                 << std::endl;
+
+    for (NuTo::IntegrationTypeBase *intType : intTypesBrick3D) {
+        std::cout << "Int Points " << intType->GetNumIntegrationPoints()
+                  << "Result: " << integrate3D(polyOrder, *(intType)) << std::endl;
```

Comment:
I second the motion for an actual test, rather than just printing the results. Also, this test should be splitted into unittests.